File: NetFramework/TreeView/VB/BoundModeITypedList/Employees/Employee.vb

Imports System.ComponentModel

Public Class Employee
    Private _id As Integer
    Private _chiefID As Integer
    Private _firstName As String
    Private _lastName As String
    Private _post As String

    Public Sub New()
    End Sub

    Public Sub New(id As Integer, chiefID As Integer, firstName As String, lastName As String, post As String)
        _id = id
        _chiefID = chiefID
        _firstName = firstName
        _lastName = lastName
        _post = post
    End Sub

    <Browsable(False)>
    Public Property ID() As Integer
        Get
            Return _id
        End Get
        Set
            If _id <> Value Then
                _id = Value
                OnPropertyChanged(New PropertyChangedEventArgs("ID"))
            End If
        End Set
    End Property
    <Browsable(False)>
    Public Property ChiefID() As Integer
        Get
            Return _chiefID
        End Get
        Set
            If _chiefID <> Value Then
                _chiefID = Value
                OnPropertyChanged(New PropertyChangedEventArgs("ChiefID"))
            End If
        End Set
    End Property
    Public Property Post() As String
        Get
            Return _post
        End Get
        Set
            If _post <> Value Then
                _post = Value
                OnPropertyChanged(New PropertyChangedEventArgs("Post"))
            End If
        End Set
    End Property
    Public Property LastName() As String
        Get
            Return _lastName
        End Get
        Set
            If _lastName <> Value Then
                _lastName = Value
                OnPropertyChanged(New PropertyChangedEventArgs("LastName"))
            End If
        End Set
    End Property
    Public Property FirstName() As String
        Get
            Return _firstName
        End Get
        Set
            If _firstName <> Value Then
                _firstName = Value
                OnPropertyChanged(New PropertyChangedEventArgs("FirstName"))
            End If
        End Set
    End Property

#Region "INotifyPropertyChanged Members"

    Public Event PropertyChanged As PropertyChangedEventHandler

    Protected Overridable Sub OnPropertyChanged(e As PropertyChangedEventArgs)
        RaiseEvent PropertyChanged(Me, e)
    End Sub

#End Region
End Class

File: NetFramework/TreeView/CS/BoundModeITypedList/Employees/Employee.cs
using System.ComponentModel;

namespace SamplesData
{
    public class Employee
    {
        private int _id;
        private int _chiefID;
        private string _firstName;
        private string _lastName;
        private string _post;

        public Employee()
        { }

        public Employee(int id, int chiefID, string firstName, string lastName, string post)
        {
            _id = id;
            _chiefID = chiefID;
            _firstName = firstName;
            _lastName = lastName;
            _post = post;
        }

        [Browsable(false)]
        public int ID
        {
            get { return _id; }
            set
            {
                if (_id != value)
                {
                    _id = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("ID"));
                }
            }
        }
        [Browsable(false)]
        public int ChiefID
        {
            get { return _chiefID; }
            set
            {
                if (_chiefID != value)
                {
                    _chiefID = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("ChiefID"));
                }
            }
        }
        public string Post
        {
            get { return _post; }
            set
            {
                if (_post != value)
                {
                    _post = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("Post"));
                }
            }
        }
        public string LastName
        {
            get { return _lastName; }
            set
            {
                if (_lastName != value)
                {
                    _lastName = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("LastName"));
                }
            }
        }
        public string FirstName
        {
            get { return _firstName; }
            set
            {
                if (_firstName != value)
                {
                    _firstName = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("FirstName"));
                }
            }
        }               

        #region INotifyPropertyChanged Members

        public event PropertyChangedEventHandler PropertyChanged;

        protected virtual void OnPropertyChanged(PropertyChangedEventArgs e)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, e);
            }
        }

        #endregion
    }
}
